web/styles/message_row.css

```diff
@@ -471,15 +471,66 @@ $time_column_min_width: 42px; /* + padding */
        for details on how this works */
     .message_row.unread {
         .date_unread_marker {
-            display: none;
+            .unread-marker-fill {
+                opacity: 0;
+            }
         }
     }
 
     /* Select all but the first .message_row.unread,
        and remove the properties set from the previous rule. */
     .message_row.unread ~ .message_row.unread {
         .date_unread_marker {
+            .unread-marker-fill {
+                opacity: 1;
+            }
+        }
+    }
+
+    /* Don't apply mention background color message date row if the previous
+       message don't have user mentioned. */
+    .message_row:not(.direct_mention) + .message_row.direct_mention {
+        /* Since message left border is 1px and unread marker is 2px,
+            we leave 1px transparent so that the border is visible. */
+        .date_unread_marker {
+            background: linear-gradient(
+                to right,
+                transparent,
+                var(--color-background-stream-message-content) 50% 100%
+            );
```

Comment:
If we go with this design, I think conceptually what we probably want to be doing is having the row start with the base color and have a `.message_row.direct_mention + .message_row.direct_mention` rule to make the date row have the mention color, if that makes sense?

Reply:
Yeah, that does makes sense to do. This PR feels more like a hack.